secure_message/services/party_service.py

```diff
@@ -8,42 +10,41 @@
 class PartyService:
 
     def __init__(self):
-        self._users_cache = {}
-        self._business_details_cache = {}
+        self._users_cache = []
 
-    def get_business_details(self, ru):
+    @staticmethod
+    def get_business_details(rus):
         """Retrieves the business details from the party service"""
-
-        ru_dict = self._business_details_cache.get(ru)
-
-        if ru_dict is None:
-            logger.info(f"Party Service: retrieve party ru data for {ru}")
-
-            party_data = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/businesses/id/{ru}",
-                                      auth=current_app.config['BASIC_AUTH'], verify=False)
-            if party_data.status_code == 200:
-                ru_dict = party_data.json()
-                self._business_details_cache[ru] = ru_dict
-                logger.debug(f"Party data retrieved for ru:{ru}")
-            else:
-                logger.debug(f"Party data failed for ru:{ru}", status_code=party_data.status_code, text=party_data.text,
-                             ru=ru)
-
-        return ru_dict
+        ru_list = []
+        payload = urlencode([("id", x) for x in rus])
+
+        party_data = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/businesses",
+                                  auth=current_app.config['BASIC_AUTH'], verify=False, params=payload)
+        if party_data.status_code == 200:
+            ru_list = party_data.json()
+            logger.debug(f"Party data retrieved for ru:{rus}")
+        else:
+            logger.debug(f"Party data failed for ru:{rus}", status_code=party_data.status_code, text=party_data.text,
+                         ru=rus)
+        return ru_list
 
     def get_user_details(self, uuid):
-
-        user_dict = self._users_cache.get(uuid)
-
-        if user_dict is None:
-            logger.info(f"Party Service: retrieve party user data for {uuid}")
-            party_data = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/respondents/id/{uuid}",
-                                      auth=current_app.config['BASIC_AUTH'], verify=False)
+        """Retrieves the user details from the party service"""
+        user_list = None
+        if self._users_cache:
```

Comment:
I'm not sure this'll do what you think it does.  I think it'll cache it for THIS app, but no others.  When you've got multiple instances running then it'll hit this cache every so often.
Ideally this should be stored in redis